### ISSUE TITLE
W1b: `Boxes.to_tensor('vertices')` fed to the `kornia.geometry.bbox` free functions reads one pixel larger per axis

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4009]
`Boxes` stores boxes inclusively (`vertices_plus`) and its exclusive exports (`xyxy`, `xywh`, `vertices`) add the pixel back on the way out — internally consistent. But the free functions in `kornia.geometry.bbox` are hard-wired inclusive with **no `mode` argument**, so the natural torchvision-compatible export, `to_tensor('vertices')`, silently gains one pixel per axis the moment it touches `infer_bbox_shape`, `validate_bbox`, `bbox_to_mask` or `transform_bbox`.

## Repro

```python
import torch
from kornia.geometry.boxes import Boxes
from kornia.geometry.bbox import infer_bbox_shape

b = Boxes.from_tensor(torch.tensor([[[1., 1., 3., 2.]]]), mode="xyxy")   # COCO/torchvision terms: w=2, h=1
print("stored:", b.data.tolist())
print("infer_bbox_shape(to_tensor('vertices')):     ", [t.tolist() for t in infer_bbox_shape(b.to_tensor("vertices")[0])])
print("infer_bbox_shape(to_tensor('vertices_plus')):", [t.tolist() for t in infer_bbox_shape(b.to_tensor("vertices_plus")[0])])
print("get_boxes_shape:", [t.tolist() for t in b.get_boxes_shape()])
```

## Observed

```
stored: [[[[1.0, 1.0], [2.0, 1.0], [2.0, 1.0], [1.0, 1.0]]]]
infer_bbox_shape(to_tensor('vertices')):      [[2.0], [3.0]]     <- h, w: one too large each
infer_bbox_shape(to_tensor('vertices_plus')): [[1.0], [2.0]]     <- correct
get_boxes_shape: [[[1.0]], [[2.0]]]                              <- correct
```

## Expected

One of:

1. The free functions accept a `mode` (default `vertices_plus`, the current behavior) so an exclusive export can be declared — no change for existing callers.
2. The free functions stay inclusive and the `Boxes` convention block states, once, "feed the free functions the `vertices_plus` export, never `vertices`" (the batch-4 docs do this regardless).

Resolved together with #3934: whichever arithmetic the window settles on, the two modules must agree on it.

## Classification

**Semantic contract** across two modules; nothing computes a wrong number under its own convention. Idiom 5 (a precondition — inclusive input — that is documented nowhere and validated nowhere) and idiom 3 (the free functions' implicit contract does not match what `Boxes` exports by default).

## Disposition

**MERGE IN WINDOW — behavior change.** Canonical: the `Boxes` mode semantics (`vertices` exclusive, `vertices_plus` inclusive). Shape: align contract, both names kept — the free functions either gain a `mode` or are documented as `vertices_plus`-only. Divergence today (executed): `infer_bbox_shape` returns `(2, 3)` for the `vertices` export of a box whose true extent is `(1, 2)`. This is the behavior-change discussion CONTRIBUTING asks for, and the decision is: **in the coordinated Repair & Deprecation window, not before (together with #3934). A PR that aliases or consolidates these ahead of it will be closed and pointed here.**

## Environment

kornia 0.9.0rc1 @ `0685b1f5` (main), torch 2.9.1, python 3.11, cpu. Found during the bbox/boxes conventions audit (batch 4, tracking issue #4020); the batch documents the behavior below as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5).

**Comment by ducha-aiki** [comment]
Scope correction, executed against `d7d6a2f6` while reviewing #4060 (which documents this issue): the opening sentence lists four consumers that read `to_tensor('vertices')` one pixel larger, but only two of them do.

- `infer_bbox_shape` — inclusive (`+1` on fixed vertex indices). In scope.
- `bbox_to_mask` — inclusive (fills through the bottom-right vertex's row and column with `<=`). In scope.
- `validate_bbox` — **invariant**: its `+1` terms sit on both sides of the same comparison and cancel, so it returns the same bool for `'vertices'` and `'vertices_plus'` input (0 mismatches over 20 000 random quadrilaterals against an exclusive reimplementation). Not in scope; the same correction was made on #3934.
- `transform_bbox` — **exclusive** on its `'xywh'` path (`xmax = xmin + width`) and no `±1` anywhere; a `'vertices'` export passes through it unchanged. Not in scope.

#4060's `Boxes` convention block classifies all five free functions from the code (`nms` is the fifth, exclusive, tracked in #4008); this comment keeps the issue from sending the repair to two functions with nothing to change.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by ducha-aiki** [comment]
The 3D container has the same trap, executed 2026-09-05 at `1eea5835`: `Boxes3D.from_tensor([[1, 2, 3, 5, 5, 8]], "xyzxyz")` has `get_boxes_shape() == (5, 3, 4)`, and `infer_bbox_shape3d(boxes.to_tensor("vertices"))` reads `(6, 4, 5)` while `to_tensor("vertices_plus")` reads `(5, 3, 4)`. Batch 4c states it on the `Boxes3D` anchor and `infer_bbox_shape3d`, and pins it as `test_wart_vertices_export_is_exclusive_for_inclusive_bbox3d_consumers_4009`. Scope of the window repair: both classes.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
